glommio/src/read_write_lock.rs

```diff
@@ -0,0 +1,1214 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+
+//! Read-write locks.
+//!
+//! Provides functionality similar to the ['std::sync::RwLock'] except that lock can not be poisoned.
```

Comment:
Should we call it `RwLock` for consistency ?

Reply:
:heavy_plus_sign: :100:

Reply:
In our previous conversation, about methods which fully read and write from the disk you suggested to use a bit different names to do not have confusion with futures API. I supposed that it is good to keep this strategy and call this lock as  ReadWriteLock instead of RwLock. But I am pro for any choices. That depends on the overall strategy of the creation of API. So at your choice. What do you think is better to distinguish names which are used in std library or use similar ones?

Reply:
I think those are different scenarios.
One thing is to have methods that exist in general traits having the same name but doing different things: that should definitely be different.

The other is to have different classes which serves the same purpose but are different in implementation.